app/questionnaire/placeholder_parser.py

```diff
@@ -24,35 +27,41 @@ def __init__(
         self._list_item_id = list_item_id
         self._location = location
         self._transformer = PlaceholderTransforms(language)
+        self._placeholder_map = {}
 
-    def _lookup_answer(self, answer_id, list_item_id=None):
-        answer = self._answer_store.get_answer(answer_id, list_item_id)
-
-        if answer:
-            return answer.value
+    def __call__(self, placeholder_list: Sequence[Mapping]) -> Mapping:
+        for placeholder in placeholder_list:
+            try:
+                self._placeholder_map[placeholder['placeholder']]
```

Comment:
Is this just to check if the placeholder is already in _placeholder_map? If so, it would be clearer replace the try/except with: 

```python
if placeholder['placeholder'] not in self._placeholder_map
```